src/blockchain.rs

```diff
@@ -46,10 +46,3 @@ pub fn is_valid_chain(chain: & Blockchain) -> bool{
 	}
 	return true
 }
-
-pub fn resolve<'a>(curr_chain: &'a Blockchain, new_chain: &'a Blockchain) -> &'a Blockchain {
```

Comment:
The reason this doesn't work is because there was no way for the compiler to figure out what &'a really meant.
This function is saying: "the result lives as long as curr_chain and new_chain does", what this means is that when we had a new potential chain, that lifetime was a new one, being shorter lived than `'a` in `Node`, and effectively shorter lived than `'a` in `resolve`.